packages/flutter/lib/src/material/input_border.dart

```diff
@@ -0,0 +1,346 @@
+  ShapeBorder lerpTo(ShapeBorder b, double t) {
+    if (b is UnderlineInputBorder) {
+      return new UnderlineInputBorder(
+        borderSide: BorderSide.lerp(borderSide, b.borderSide, t),
+      );
+    }
+    return super.lerpTo(b, t);
+  }
+
+  /// Draw a horizontal line at the bottom of [rect].
+  ///
+  /// The [borderSide] defines the line's color and weight.
+  @override
+  void paint(Canvas canvas, Rect rect, { TextDirection textDirection }) {
+    canvas.drawLine(rect.bottomLeft, rect.bottomRight, borderSide.toPaint());
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (runtimeType != other.runtimeType)
+      return false;
+    final InputBorder typedOther = other;
+    return typedOther.borderSide == borderSide;
+  }
+
+  @override
+  int get hashCode => borderSide.hashCode;
+}
+
+/// Draws a rounded rectangle around an [InputDecorator]'s container.
+///
+/// When the input decorator's label is floating, for example because its
+/// input child has the focus, the label appears in a gap in the border outline.
+///
+/// The input decorator's "container" is the optionally filled area above the
+/// decorator's helper, error, and counter.
+///
+/// See also:
+///
+///  * [UnderlineInputBorder], the default [InputDecorator] border which
+///    draws a horizontal line at the bottom of the input decorator's container.
+///  * [InputDecoration], which is used to configure an [InputDecorator].
+class OutlineInputBorder extends InputBorder {
+  /// Creates a rounded rectangle outline border for an [InputDecorator].
+  ///
+  /// The [borderSide] parameter defaults to [BorderSide.none] (it must not be
+  /// null). Applications typically do not specify a [borderSide] parameter
+  /// because the input decorator substitutes its own, using [copyWith], based
+  /// on the current theme and [InputDecorator.isFocused].
+  ///
+  /// If [borderRadius] is null (the default) then the border's corners
+  /// are drawn with a radius of 4dps. The corner radii must be circular, i.e.
+  /// their [Radius.x] and [Radius.y] values must be the same.
+  const OutlineInputBorder({
+    BorderSide borderSide: BorderSide.none,
+    BorderRadius borderRadius,
+    this.gapPad: 4.0,
+  }) : assert(gapPad != null && gapPad >= 0.0),
+       _borderRadius = borderRadius,
+       super(borderSide: borderSide);
+
+  /// Horizontal padding on either side of the border's
+  /// [InputDecoration.labelText] width gap.
+  ///
+  /// This value is used by the [paint] method to compute the actual gap width.
+  final double gapPad;
+
+  /// The radii of the border's rounded rectangle corners.
+  ///
+  /// The corner radii must be circular, i.e. their [Radius.x] and [Radius.y]
+  /// values must be the same.
+  BorderRadius get borderRadius => _borderRadius ?? new BorderRadius.circular(4.0);
+  final BorderRadius _borderRadius;
+
+  @override
+  OutlineInputBorder copyWith({ BorderSide borderSide }) {
```

Comment:
for completeness, also add borderRadius and gapPad

Reply:
Done.